src/cmd/flux-mini.py

```diff
@@ -1773,12 +1774,34 @@ def __init__(self, **kwargs):
             help="Batch script and arguments to submit",
         )
 
-    @staticmethod
-    def read_script(args):
+    def parse_directive_args(self, name, batchscript):
+        """
+        Parse any directives in batchscript.directives, then apply
+        command line arguments in self.argv. This allows command line
+        to override file directives
+        """
+        args = None
+        for item in batchscript.directives:
+            try:
+                if item.action == "SETARGS":
+                    args = self.parser.parse_args(item.args, namespace=args)
+            except SystemExit:
+                #  Argparse exits on error. Give the user a clue
+                #  about which line failed in the source file:
+                LOGGER.error(f"argument parsing failed at {name} line {item.lineno}")
+                sys.exit(2)
+        args = self.parser.parse_args(self.argv, namespace=args)
+        return batchscript.script, args
+
+    def process_script(self, args):
+        """
+        Process a batch script that may contain RFC 36 directives.
+        Returns the ingested script and new argparse args Namespace.
+        """
         if args.SCRIPT:
             if args.wrap:
-                #  Wrap args in /bin/sh script
-                return "#!/bin/sh\n" + " ".join(args.SCRIPT) + "\n"
+                #  Return script which will be wrapped by caller
+                return " ".join(args.SCRIPT) + "\n", args
```

Comment:
Hm, I think the diff makes this a little confusing. The change is not independent, but is required because the function now processes a batch script which may contain directives, but if the `--wrap` option was given ,then the script cannot have directives since it was provided on the command line. So we just return. This check could be moved out a function into `main()`, but it seemed better to abstract the "script processing" into a single function.

Does that make sense?

Reply:
Ahh ok makes sense